doc/continuous_integration/README.md

```diff
@@ -0,0 +1,67 @@
+# Continuous Integration Documentation
+Documentation on the existing CI for the project is resides here.
+
+## Overview
+The existing CI is composed of multiple integration services that together help provide maintainers a fast and scalable testing environment. To help detect upstream breakages quickly as well, the existing CI allows for changes to be evaluated using the latest development dependencies. In light of the large dependency footprint a high-level ROS2 navigation stack necessitates, the use of each integration service is optimized to maximize caching of environmental setup and increase workflow throughput. As these optimizations add complexity to the CI configuration, this documentation provides further explanations and reasoning behind each configuration.
+
+![pipeline](figs/pipeline.svg)
```

Comment:
> The "-devel" branch statement isn't really true, currently CI is broken for all the -devel branches from the master->main conversion.

The CI architecture design this documents affords for this, it just hasn't been used.
If you'd like to enable it, we can open a separate PR.

> There's an arrow "Commit pushed to master branch" that points to "ros-planning/navigation2 nightly build" that I don't think is correct, since the nightly builds are on a timer and not triggered by pushing to master.

Good catch, that can be removed, given we've now disabled `Autobuild` for the build rules on Dockerhub.

> I don't think (correct me) that the caches for release and debug are the same.

The caches for release and debug jobs in the same workflow don't collide thanks to the different prefix in the cache key names. I tried adding two arrows to make that separation more apparent. Also, corrected the fact that test jobs don't reuse caches from prior test jobs, only from build jobs from the same workflow.

3b2906ebe0588c19017484467d8fb225ecf63430